### ISSUE TITLE
Don't know how to parse column type: 816425

### ISSUE THREAD

**Issue by grantcarthew** [denodrivers/postgres#128]
Here is the DDL for the table that was being queried at the :

```sql
    CREATE TABLE schemaname.tablename (
      id text PRIMARY KEY NOT NULL DEFAULT 'a-key-is-here',
      schemaVersion INTEGER NOT NULL DEFAULT(0),
      status custom.enum NOT NULL DEFAULT('ready'),
      updatedAt TIMESTAMP WITH TIME ZONE,
      updatedBy text NOT NULL
    )
```

**Comment by zarszz** [comment]
i also faced this error when declaring a some custom type like this 
```typescript
import { client } from "../../helpers/database.ts";

await client.connect();
await client.query(`
    --create types
    DO $$
    BEGIN
        IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'user_role') THEN
            CREATE TYPE user_role AS ENUM
            (
                'student', 'admin', 'lecturer'
            );
        END IF;
        --more types here...
    END$$;
`)
await client.query(`
    CREATE TABLE users(
        email VARCHAR(30) NOT NULL,
        password VARCHAR(50) NOT NULL,
        student_id VARCHAR(13) NOT NULL,
        role user_role,
        FOREIGN KEY (student_id) REFERENCES student(student_id)
    )
`)

await client.end();
```
my query
```typescript
        await client.connect();
        // student_id has string type and successfully querying in sql
        const userIsRegistered = await client.query(
            `SELECT * FROM users WHERE student_id=$1`,
            student_id
        );
```

and throwing a error like this

```shell
Error: Don't know how to parse column type: 18405
    at decodeText (https://deno.land/x/postgres/decode.ts:228:13)
    at decode (https://deno.land/x/postgres/decode.ts:236:12)
    at QueryResult._parseDataRow (https://deno.land/x/postgres/query.ts:35:24)
    at QueryResult.handleDataRow (https://deno.land/x/postgres/query.ts:47:28)
    at Connection._preparedQuery (https://deno.land/x/postgres/connection.ts:504:18)
    at async Connection.query (https://deno.land/x/postgres/connection.ts:528:12)
    at async Client.query (https://deno.land/x/postgres/client.ts:24:12)
    at async register (file:///home/myDomain/Documents/deno/http/oak-lib/students-api/controllers/authController.ts:50:34)
    at async allowedMethods (https://deno.land/x/oak/router.ts:259:7)
    at async Application.#handleRequest (https://deno.land/x/oak/application.ts:190:9)
```

**Comment by jakajancar** [comment]
If you want user-defined types and are not too deeply invested in deno-postgres, you can give [pgc4d](https://github.com/jakajancar/pgc4d) a go.

As perhaps an inspiration  for deno-postgres, [here's how I implemented it](https://github.com/jakajancar/pgc4d/blob/master/src/data_type_registry.ts).

**Comment by grantcarthew** [comment]
That project looks great @jakajancar. I'm going to open an issue there about types. Thanks for the input.

**Comment by Soremwar** [comment]
FYI I intend to resolve this by allowing users to register their own handlers on the client for specific or unknown column types before 1.0

However before that, I intend to solve #184 and then default all new types to strings, which might be as useful (though not as nice) for users

**Comment by Soremwar** [comment]
This was solved on v0.7.0

**Comment by bombillazo** [comment]
hey, any guide on how to add a transformation/parser using this pg driver ? I want queries returning a `numeric` type to be parsed as float instead of string
